org.lflang/src/org/lflang/generator/GeneratorBase.xtend

```diff
@@ -284,21 +284,47 @@ abstract class GeneratorBase extends AbstractLFValidator {
      * 
      */
     protected def void setTargetConfig(IGeneratorContext context) {
-        // If there are any physical actions, ensure the threaded engine is used.
-        for (action : fileConfig.resource.allContents.toIterable.filter(Action)) {
-            if (action.origin == ActionOrigin.PHYSICAL) {
-                targetConfig.threads = 1
-            }
-        }
 
         val target = fileConfig.resource.findTarget
         if (target.config !== null) {
             // Update the configuration according to the set target properties.
             TargetProperty.set(this.targetConfig, target.config.pairs ?: emptyList)
         }
-
-        // Override target properties if specified as command line arguments.
-        if (context instanceof StandaloneContext) {
+        // If there are any physical actions, ensure the threaded engine is used and that
+        // keepalive is set to true, unless the user has explicitly set it to false.
+        for (action : fileConfig.resource.allContents.toIterable.filter(Action)) {
+            if (action.origin == ActionOrigin.PHYSICAL) {
+                // If the unthreaded runtime is requested, use the threaded runtime instead
+                // because it is the only one currently capable of handling asynchronous events.
+                if (targetConfig.threads < 1) {
+                    targetConfig.threads = 1
+                    errorReporter.reportWarning(
+                        target, 
+                        '''Using the threaded C runtime to allow for asynchronous handling of«
+                        » physical action «action.name».'''
+                    );
+                }
+                // Check if the user has explicitly set keepalive to false or true
+                if (target.config?.pairs?.filter[
```

Comment:
I don't think this is the right place to do this. I would just add a member `keepAliveSpecified` or something like that to `TargetConfig` that indicates as to whether the property was set explicitly.

Reply:
See below.